src/consumer.cpp

```diff
@@ -15,9 +15,15 @@ void chimera::Consumer::HandleTranslationUnit(ASTContext &context)
 {
     // Use the current translation unit to resolve the YAML configuration.
     chimera::Configuration &config = chimera::Configuration::GetInstance();
-    chimera::Visitor visitor(ci_, config.Process(ci_));
+
+    std::shared_ptr<chimera::CompiledConfiguration> compiled_config
+        = config.Process(ci_);
```

Comment:
Does this even need to be a pointer anymore?  We can just pass by reference into the visitor, because we know it will stay in scope over the lifetime of the visitor, right?
```suggestion
chimera::CompiledConfiguration compiled_config = config.Process(ci_);
```

Reply:
The scope seems okay, but in order to make `Configuration::Process()` to return a value of `CompiledConfiguration`, we have to un-delete copy constructor (and/or move constructor) of `CompiledConfiguration`, which is currently deleted.

We could either (1) implement the necessary methods, (2) create `unique_ptr`  and pass dereferenced value, or (3) use the current implementation in this PR. Which one would you prefer?

Reply:
Let's do (2), that is the most similar to the existing behavior.

Reply:
Done in https://github.com/personalrobotics/chimera/pull/269/commits/d68eceeb3cab3d1efeee3356faefa7f7e2c1b547